### ISSUE TITLE
[Feature Request] Option to Use Git Configured Editor as Default External Editor

### ISSUE THREAD

**Issue by kevinmatthes** [AmmarAbouZor/tui-journal#38]
Many users can be expected to have Git configured.  Thus, it is very likely that, if Git is configured, also a preferred editor is set.  With the `git2` crate (MIT OR Apache-2.0), you can access this configuration and offer to use this editor as the default external editor.  If the configuration should not be accessible for some reason, e.g., the current working directory is not a Git repository, Git is not configured, or a preferred editor for Git is not set, the current implementation should be applied as fallback.  Thus, the configured Git editor would be an additional offer to the users.

For the implementation, first of all, run `cargo add git2 --no-default-features` to add the crate with minimal dependencies to the Rust manifest.  Then, to access the configuration, you can use the following function.

```rust
fn get_git_editor() -> Result<String, u8> {
  git2::Repository::open(".").map_or(Err(64),
    |r| {
      r.config().map_or(Err(69), |c| c.get_string("core.editor").map_or(Err(65), Ok))
    },
  )
}
```

The function returns you either the configured Git editor or an integer according to the sysexits header; the latter is meant to be a placeholder for your preferred return value in case of an error.  For the real Rust implementation of sysexits, you can use https://github.com/sorairolake/sysexits-rs (`cargo add sysexits`).

What do you think about this feature?

**Comment by AmmarAbouZor** [comment]
This could be useful for cross platform compatibility, since the current implementation has only UNIX in mind. I'll try to implement this and see how it would work

**Comment by AmmarAbouZor** [comment]
@kevinmatthes  I've found the method `git2::Config::open_default()` which opens the git global the configurations. 

I think it's more useful this way since the users most likely set the editor globally and this way we are not tied to the current directory. 
If you think that there are other benefits from using the current directory please let me know

**Comment by kevinmatthes** [comment]
@AmmarAbouZor

Thanks for pointing out the access to the global configuration!  I did only know about the local one, by now.

One thing that might be considerable is the ability to configure the preferred Git editor per repository.  This is especially useful for users who contribute to multiple text editors.  These users may have set their favourite editor in the global settings and altered their preferred editor per repository by configuring the respective editor the corresponding project is about.  This use case should rather not be underestimated as there are plenty of text editors, especially in the Rust ecosystem where implementing them is very easy; furthermore, I experienced myself that contributors to text editors usually participate in the development of multiple editors (or at least switch to another project when one should become discontinued whilst still the setting regarding the Git editor for the discontinued project is reasonable as the maintainer might decide to work on their project, again).

However, this is a rather exotic edge case as most users cannot be expected to contribute to more than one editor, if any.  So, it is up to you how you decide to take this scenario into account for the further development since the current implementation should fit for about 95% of the use cases, I guess.
